src/hiprtc.cpp

```diff
@@ -286,22 +300,33 @@ struct _hiprtcProgram {
         return true;
     }
 
+    void replaceExtension(std::string& fileName, const std::string &ext) const {
+        auto res = std::find(fileName.rbegin(), fileName.rend(), '.');
+        auto sloc = std::find(fileName.rbegin(), fileName.rend(), '/');
+        if(res == fileName.rend() || res > sloc) {
```

Comment:
Could you replace these with `std::string::rfind` instead?  It would make the code a lot more readable, especially for the non-obvious usages of `reverse_iterators` in the `res > sloc` comparison and in `replace` below.

Reply:
Done